util/generate-network-configurations.py

```diff
@@ -0,0 +1,161 @@
+from troposphere import Template, Parameter, Ref, Output, Equals, Not, GetAtt, If
+from troposphere.ec2 import (
+    Subnet,
+    InternetGateway,
+    RouteTable,
+    VPCGatewayAttachment,
+    Route,
+    SubnetRouteTableAssociation,
+    NatGateway,
+    EIP,
+)
+
+
+class NetworkHandler:
+    def __init__(self):
+        self.template = Template()
+
+    @staticmethod
+    def __add_name(resource_name):
+        return [{"Key": "Name", "Value": f"Pcluster {resource_name}"}]
+
+    def __add_parameter(self, name, description, expected_input):
+        return self.template.add_parameter(Parameter(name, Description=description, Type=expected_input))
+
+    def __add_output(self, name, description, value):
+        return self.template.add_output(Output(name, Description=description, Value=value))
+
+    def __write_template(self, path):
+        with open(path, "w+") as file:
+            file.write(self.template.to_yaml())
+
+    def __create_gateway(self, vpc, gateway_attachment_name):
+        internet_gateway = self.template.add_resource(
+            InternetGateway("InternetGateway", Tags=NetworkHandler.__add_name("Internet Gateway"))
+        )
+        self.template.add_resource(
+            VPCGatewayAttachment(gateway_attachment_name, VpcId=Ref(vpc), InternetGatewayId=Ref(internet_gateway))
+        )
+        return internet_gateway
+
+    def __create_route_table_with_ig(self, vpc, internet_gateway, gateway_attachment_name, subnet):
+        route_table = self.template.add_resource(
+            RouteTable("RouteTable", VpcId=Ref(vpc), Tags=NetworkHandler.__add_name("Public Route Table"))
+        )
+        self.template.add_resource(
+            Route(
+                "Route",
+                DependsOn=gateway_attachment_name,
+                GatewayId=Ref(internet_gateway),
+                DestinationCidrBlock="0.0.0.0/0",
+                RouteTableId=Ref(route_table),
+            )
+        )
+        self.template.add_resource(
+            SubnetRouteTableAssociation(
+                "SubnetRouteTableAssociation", SubnetId=Ref(subnet), RouteTableId=Ref(route_table)
+            )
+        )
+        return route_table
+
+    def __create_nat(self, compute_subnet, condition, master_subnet, vpc):
+        nat_eip = self.template.add_resource(EIP("NatEip", Domain=Ref(vpc), Condition=condition))
+        nat = self.template.add_resource(
+            NatGateway(
+                "Nat",
+                AllocationId=GetAtt(nat_eip, "AllocationId"),
+                SubnetId=Ref(master_subnet),
+                Condition=condition,
+                Tags=[{"Key": "Name", "Value": "NAT"}],
```

Comment:
We can use a more explicative name for the resources we are creating. E.g. NAT --> pcluster-NAT ?